packages/next/src/server/lib/render-server.ts

```diff
@@ -143,6 +144,6 @@ export async function initialize(opts: {
         return reject(err)
       }
     })
-    server.listen(await getFreePort(), '0.0.0.0')
+    server.listen(await getFreePort(), hostname)
```

Comment:
Changing this to hostname seems like it could cause the issue you're describing, `0.0.0.0` should ensure it's available to be invoked via `127.0.0.1`.

Reply:
@ijjk no, it won't. I tried it myself. Rather than using the provided hostname for the proxy server, we now use what `server.address()` returns. So if the provided host is `localhost`, it will return `::1` if IPv6 is used and `127.0.0.1` if IPv4 is used, ensuring that the hostname for the proxy server is always correct.

Reply:
Is there a quick way to install this particular change under `node_modules` in my own app repository? I could test out if this change resolves the issue I'm getting here: https://github.com/vercel/next.js/issues/52215

Reply:
@michaelangrivera I don't know, maybe try `npm install 'https://gitpkg.now.sh/DuCanhGH/next.js/packages/next?00457da5f319d9ce9f49f53ba8c80779c502906a'`?

Reply:
@michaelangrivera eh, doesn't seem possible without straight up publishing a fork of Next.js...